lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2495,7 +2530,10 @@ SubExprInfo ExprEmitter::emitSubExpr(Value exp,
     }
   // Okay, this is an expression we should emit inline.  Do this through our
   // visitor.
+  bool savedAssignmentLikeContext = this->isAssignmentLikeContext;
+  this->isAssignmentLikeContext = isAssignmentLikeContext;
   auto expInfo = dispatchCombinationalVisitor(exp.getDefiningOp());
+  this->isAssignmentLikeContext = savedAssignmentLikeContext;
```

Comment:
Can maybe use `llvm::SaveAndRestore` here

Reply:
Great idea! I was looking for an RAII way to do this but couldn't remember what the name of the thing was :see_no_evil:. Updated.